aries_cloudagent/indy/sdk/profile.py

```diff
@@ -116,6 +122,19 @@ async def close(self):
             await self.opened.close()
             self.opened = None
 
+    def _make_finalizer(self) -> finalize:
+        """Return a finalizer for this profile.
+
+        See docs for weakref.finalize for more details on behavior of finalizers.
+        """
+
+        def _finalize(opened: Optional[IndyOpenWallet]):
+            if opened:
```

Comment:
I don't think opened can be None because it receives the original IndyOpenWallet instance. Maybe `if opened.handle`?

Reply:
This came as a recommendation from pyright which indicates that IndySdkProfile.opened is `IndyOpenWallet | None`. I think it derived this from the `self.opened = None` line in the `close` method.

Following that same vein, I think it was possible before for the finalizer to be created after `close` had been called. Now that it's only intended to be called in `__init__`, I don't think this is a problem any more. I'll fix that.